nni/mutable/symbol.py

```diff
@@ -0,0 +1,472 @@
+        return self.expr_cls(operator.rshift, '{} >> {}', [self, other])
+
+    def __rrshift__(self, other: Any) -> Any:
+        return self.expr_cls(operator.rshift, '{} >> {}', [other, self])
+
+    def __and__(self, other: Any) -> Any:
+        return self.expr_cls(operator.and_, '{} & {}', [self, other])
+
+    def __rand__(self, other: Any) -> Any:
+        return self.expr_cls(operator.and_, '{} & {}', [other, self])
+
+    def __xor__(self, other: Any) -> Any:
+        return self.expr_cls(operator.xor, '{} ^ {}', [self, other])
+
+    def __rxor__(self, other: Any) -> Any:
+        return self.expr_cls(operator.xor, '{} ^ {}', [other, self])
+
+    def __or__(self, other: Any) -> Any:
+        return self.expr_cls(operator.or_, '{} | {}', [self, other])
+
+    def __ror__(self, other: Any) -> Any:
+        return self.expr_cls(operator.or_, '{} | {}', [other, self])
+
+    def __lt__(self, other: Any) -> Any:
+        return self.expr_cls(operator.lt, '{} < {}', [self, other])
+
+    def __le__(self, other: Any) -> Any:
+        return self.expr_cls(operator.le, '{} <= {}', [self, other])
+
+    def __ne__(self, other: Any) -> Any:
+        return self.expr_cls(operator.ne, '{} != {}', [self, other])
+
+    def __ge__(self, other: Any) -> Any:
+        return self.expr_cls(operator.ge, '{} >= {}', [self, other])
+
+    def __gt__(self, other: Any) -> Any:
+        return self.expr_cls(operator.gt, '{} > {}', [self, other])
+    # endregion
+
+    # __pow__, __divmod__, __abs__ are special ones.
+    # Not easy to cover those cases with codegen.
+    def __pow__(self, other: Any, modulo: Any | None = None) -> Any:
+        if modulo is not None:
+            return self.expr_cls(pow, 'pow({}, {}, {})', [self, other, modulo])
+        return self.expr_cls(lambda a, b: a ** b, '{} ** {}', [self, other])
+
+    def __rpow__(self, other: Any, modulo: Any | None = None) -> Any:
+        if modulo is not None:
+            return self.expr_cls(pow, 'pow({}, {}, {})', [other, self, modulo])
+        return self.expr_cls(lambda a, b: a ** b, '{} ** {}', [other, self])
+
+    def __divmod__(self, other: Any) -> Any:
+        return self.expr_cls(divmod, 'divmod({}, {})', [self, other])
+
+    def __rdivmod__(self, other: Any) -> Any:
+        return self.expr_cls(divmod, 'divmod({}, {})', [other, self])
+
+    def __abs__(self) -> Any:
+        return self.expr_cls(abs, 'abs({})', [self])
+
+
+class Symbol(SymbolicExpression):
+    """
+    The leaf node of a symbolic expression.
+    Each :class:`Symbol` represents one variable in the expression.
+
+    Variable with the same ``label`` share the same value.
+
+    Operations on symbols (e.g., ``a + b``) will result in a new :class:`SymbolicExpression`.
+
+    Parameters
+    ----------
+    label
+        Each symbol is bound with a label, i.e., the variable name.
+    """
+
+    def __init__(self, label: str | None) -> None:
+        self.label = label
```

Comment:
should a symbol without label exist? can we create a new symbol then a unique name is assigned to it automatically?

Reply:
It used to be useful. But I think it's unnecessary now.